client_push_go17.go

```diff
@@ -0,0 +1,50 @@
+// +build go1.7
```

Comment:
This is okay. "from Go version 1.7 onward" https://golang.org/pkg/go/build/

Though I don't know if it's needed.